test/utils/helpers.ts

```diff
@@ -196,9 +198,81 @@ export const waitForFinalizedEvent = async <T extends AnyTuple>(
         })
     );
 };
+
+/**
+ * Wait for an event to show up on chain (unfinalized).
+ *
+ * This method will only subscribe to new blocks once called and then look for a specific event.
+ * Contrary to @see {@link DefaultTransactionAPI.waitForEvent} which looks at all past events
+ * and may return due to older events using the same name.
+ *
+ * Note: This method checks the parent blocks in case the event we are looking for
+ * was finalized in a block just before this method was called.
+ *
+ * @param interBtcApi the api to use for querying
+ * @param event the event to wait for, eg. `api.events.assetRegistry.RegisteredAsset`
+ */
+export const waitForIncludedEvent = async <T extends AnyTuple>(
+    interBtcApi: InterBtcApi,
+    event: AugmentedEvent<ApiTypes, T>
+): Promise<void> => {
+    return new Promise<void>((resolve, _) =>
+        interBtcApi.system.subscribeToCurrentBlockHeads(async (header) => {
+            const events = await interBtcApi.api.query.system.events.at(header.parentHash);
+            if (eventRecordVectorContains(events, event)) {
+                resolve();
+            }
+        })
+    );
+};
+
+const eventRecordVectorContains = <T extends AnyTuple>(
+    eventRecords: Vec<FrameSystemEventRecord>,
+    event: AugmentedEvent<ApiTypes, T>
+): boolean => {
+    return eventRecords.find((record) => event.is(record.event)) !== undefined;
+};
+
+/**
+ * Wait for an event to show up on chain.
+ *
+ * This method will only subscribe to new blocks once called and then look for a specific event.
+ *
+ * Note: This method checks the parent blocks in case the event we are looking for
+ * was finalized/included in a block just before this method was called.
+ * @param interBtcApi the api to use for querying
+ * @param event the event to wait for, eg. `api.events.assetRegistry.RegisteredAsset`
+ * @param finalized [optional] true to wait for finalized event, false (default) to only wait for included event
+ * @param timeoutMs [optional] a timeout in milliseconds to wait for the event. Rejects if not found before timeout. Default: no timeout
+ * @returns a promise that resolves when the event is found, and if a timeout is provided, rejects if not found before timeout.
+ */
+export const waitForEvent = async <T extends AnyTuple>(
+    interBtcApi: InterBtcApi,
+    event: AugmentedEvent<ApiTypes, T>,
+    finalized: boolean = false,
+    timeoutMs?: number
+): Promise<boolean> => {
+    let timeoutHandle: NodeJS.Timeout;
+    const timeoutPromise =
+        timeoutMs !== undefined
+            ? new Promise<void>((_, reject) => {
+                  timeoutHandle = setTimeout(() => reject(), timeoutMs);
+              })
+            : Promise.resolve();
+
+    const waitForEventPromise = finalized
+        ? waitForFinalizedEvent(interBtcApi, event)
+        : waitForIncludedEvent(interBtcApi, event);
+
+    await Promise.race([waitForEventPromise, timeoutPromise]).then((_) => {
+        clearTimeout(timeoutHandle);
+    });
```

Comment:
Ooo this is good to know, `race` presumably returns the first `Promise` to resolve?